instruments/tests/test_tektronix/test_tekdpo70000.py

```diff
@@ -0,0 +1,2022 @@
+                     f"{n_bytes}"
+    with expected_protocol(
+            ik.tektronix.TekDPO70000,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        assert inst._dtype(binary_format, byte_order, n_bytes) == \
+               value_expected
+
+
+# DATA SOURCE - TESTED WITH CHANNELS #
+
+@pytest.mark.parametrize("channel", [it for it in range(4)])
+def test_data_source_name(channel):
+    """Query the name of a data source."""
+    with expected_protocol(
+            ik.tektronix.TekDPO70000,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        assert inst.channel[channel].name == f"CH{channel+1}"
+
+
+@pytest.mark.skip(reason="Proper binary data result in an issue when reading "
+                         "the next variable because the "
+                         "loopback_communicator does not flush the input "
+                         "properly!")
+@pytest.mark.parametrize("channel", [it for it in range(4)])
+@given(values=st.lists(st.integers(min_value=-2147483648,
+                                   max_value=2147483647), min_size=1))
+def test_data_source_read_waveform(channel, values):
+    """Read waveform from data source, binary format only!"""
+    # select one set to test for:
+    binary_format = ik.tektronix.TekDPO70000.BinaryFormat.int  # go w/ values
+    byte_order = ik.tektronix.TekDPO70000.ByteOrder.big_endian
+    n_bytes = 4
+    # get the dtype
+    dtype_set = ik.tektronix.TekDPO70000._dtype(binary_format, byte_order,
+                                                n_bytes)
+
+    # pack the values
+    values_packed = b"".join(struct.pack(dtype_set[:-1],
+                                         value) for value in values)
+    values_len = str(len(values_packed)).encode()
+    values_len_of_len = str(len(values_len)).encode()
+    values = np.array(values)
+    # scale the values
+    scale = 1.
+    position = 0.
+    offset = 0.
+    scaled_values = scale * ((ik.tektronix.TekDPO70000.VERT_DIVS / 2) *
+                             values.astype(float) / (2**15) - position
+                             ) + offset
+
+    # run through the instrument
+    with expected_protocol(
+            ik.tektronix.TekDPO70000,
+            [
+                "DAT:SOU?",  # query data source
+                "DAT:ENC FAS",  # fastest encoding
+                "WFMO:BYT_N?",  # get n_bytes
+                "WFMO:BN_F?",  # outgoing binary format
+                "WFMO:BYT_O?",  # outgoing byte order
+                "CURV?",  # query data
+                f"CH{channel + 1}:SCALE?",  # scale raw data
+                f"CH{channel + 1}:POS?",
+                f"CH{channel + 1}:OFFS?"
+
+            ],
+            [
+                f"CH{channel+1}",
+                f"{n_bytes}",
+                f"{binary_format.value}",
+                f"{byte_order.value}",
+                b"#" + values_len_of_len + values_len + values_packed,
```

Comment:
Here is the real issue in the test I would like to run. Packing the data properly and then reading them back in leaves behind a `\n`. The `flush_input` method of the `loopback_communicator` is then called, but doesn't do anything. That means that reading the scale in the next line receives a `\n` instead of a value, and then fails.
My proposed bug fix is discussed in PR comment.